apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/Internal/Logging/Logger.cs

```diff
@@ -13,172 +13,304 @@
 // limitations under the License.
 
 using System;
+using System.Collections;
+using System.Collections.Concurrent;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
 
 namespace Google.Cloud.Spanner.V1.Internal.Logging
 {
     /// <summary>
-    /// Provides diagnostic information on Cloud Spanner.
-    /// By default, diagnostic logs will be sent to Trace.WriteLine for .Net desktop
-    /// and stderr for .Net Core.
-    /// The default LogLevel is LogLevel.Error which will only log error information.
-    /// Performance logging can be turned on by setting LogPerformanceTraces = true.
-    /// TODO: examine https://github.com/grpc/grpc/tree/master/src/csharp/Grpc.Core/Logging
-    /// to determine if it should be used instead.
+    /// This is an internal property and is not intended to be used by external code.
     /// </summary>
     public abstract class Logger
     {
         private static Logger s_instance;
+        private int _perfLoggingTaskEnabled;
+        private readonly ConcurrentDictionary<string, PerformanceTimeEntry> _perfCounterDictionary
+            = new ConcurrentDictionary<string, PerformanceTimeEntry>();
+        private readonly object _perfLogSyncObject = new object();
```

Comment:
Two questions here:
1. Is `_perfLogSyncObject` needed? It seems like you could lock on `_perfCounterDictionary` instead in all cases.
2. Why does `_perfCounterDictionary` need to be a `ConcurrentDictionary<...>`? It seems to always be used inside a lock. Shouldn't `Dictionary<...>` suffice?

Reply:
I agree.  Got rid of sync and switched to a normal Dictionary.